File: Arrays/SummaryRanges/Go/228-Summary-Ranges.go
func summaryRanges(nums []int) []string {
    result := []string{}

    i := 0

    for j := 1; j< len(nums) + 1; j++ {
        if j == len(nums) || nums[j] != nums[j-1] + 1 {
            if i == j -1 {
                srange := fmt.Sprintf(\%d\, nums[i])
                result = append(result, srange)
            } else {
                srange := fmt.Sprintf(\%d->%d\, nums[i], nums[j-1])
                result = append(result, srange)
            }

            i = j
        }
    }
    return result
}

File: Arrays/SummaryRanges/Python/228-Summary-Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        res = []
        i = 0


        for j in range(1, len(nums) + 1):
            if j == len(nums) or nums[j] != nums[j-1]+1:
                if i == j - 1:
                    res.append(str(nums[i]))
                else:
                    res.append(f\{nums[i]}->{nums[j-1]}\)
                
                i = j

        return res
        